Source/WebKit/GPUProcess/media/RemoteVideoFrameObjectHeap.cpp

```diff
@@ -47,10 +47,9 @@ namespace WebKit {
 
 using namespace WebCore;
 
-static WorkQueue& remoteVideoFrameObjectHeapQueue()
+static Ref<WorkQueue> remoteVideoFrameObjectHeapQueue()
 {
-    static NeverDestroyed queue = WorkQueue::create("org.webkit.RemoteVideoFrameObjectHeap"_s, WorkQueue::QOS::UserInteractive);
-    return queue.get();
+    return WorkQueue::create("org.webkit.RemoteVideoFrameObjectHeap"_s, WorkQueue::QOS::UserInteractive);
```

Comment:
This is a pretty major change to the lifecycle management of this object! I think to fix the static analyzer issue here you want to rename this to some flavor of `fooSingleton()` and keep it a NeverDestroyed.

Reply:
Oops! You're totally right--I missed the NeverDestroyed there. Since the analyzer was compiling about `remoteVideoFrameObjectHeapQueue()` being used as a call argument but being uncounted, I changed its result to be held as a ref in `RemoteVideoFrameObjectHeap::create`. I believe that should fix the static analyzer issue. Should I still rename the function to include `singleton` or does it look ok now?

Reply:
I think if you keep everything else the same but rename the getter to be `remoteVideoFrameObjectHeapQueueSingleton()` it will be OK with you returning a raw reference. At least that's how I understood rniwa's message on webkit-dev the other day.

Reply:
I think a `Ref` is redundant since the object is static and can never be destroyed.

Reply:
Oh I see! That's pretty neat. I didn't think the analyzer would examine function names too--just return types. Thank you for letting me know. I've fixed it in the latest iteration. Is webkit-dev one of our slack channels? I'm not sure if I'm in it.

Reply:
It's the public mailing list. You can join here: https://lists.webkit.org/mailman/listinfo/webkit-dev

Reply:
Got it--thank you.